query-engine/core/src/metrics/recorder.rs

```diff
@@ -0,0 +1,121 @@
+use std::sync::Arc;
+
+use metrics::KeyName;
+use metrics::{Counter, CounterFn, Gauge, GaugeFn, Histogram, HistogramFn, Key, Recorder, Unit};
+
+use super::common::KeyLabels;
+use super::{METRIC_COUNTER, METRIC_DESCRIPTION, METRIC_GAUGE, METRIC_HISTOGRAM, METRIC_TARGET};
+
+#[derive(Default)]
+pub(crate) struct MetricRecorder;
+
+impl MetricRecorder {
+    fn register_description(&self, name: &str, description: &str) {
+        trace!(
+            target: METRIC_TARGET,
+            name = name,
+            metric_type = METRIC_DESCRIPTION,
+            description = description
+        );
+    }
+}
+
+impl Recorder for MetricRecorder {
+    fn describe_counter(&self, key_name: KeyName, _unit: Option<Unit>, description: &'static str) {
```

Comment:
Odd interface by the `Recorder` with the `Option<Unit>` 🤷

Reply:
Yeah, I think they trying to cater for all types of metrics.